tabpfn_client/client.py

```diff
@@ -231,10 +236,10 @@ def try_authenticate(self, access_token) -> bool:
         )
 
         self._validate_response(response, "try_authenticate", only_version_check=True)
-
         if response.status_code == 200:
             is_authenticated = True
-
+        elif response.status_code == 403:
+            is_authenticated = (False, self.Status.USER_NOT_VERIFIED)
```

Comment:
Sure

Reply:
I have changed the name of the function but I will have to return bool | None to distinguish whether user is not  authenticated because of user verification or outdated token.